kanidm_client/tests/default_entries.rs

```diff
@@ -0,0 +1,622 @@
+                .map(ToString::to_string)
+                .collect();
+
+        let groups = rsclient.idm_group_list().unwrap();
+        let group_names: HashSet<String> = groups
+            .iter()
+            .map(|entry| entry.attrs.get("name").unwrap().first().unwrap())
+            .cloned()
+            .collect();
+        assert_eq!(default_group_names, group_names);
+
+        test_modify_group(&rsclient, &DEFAULT_HP_GROUP_NAMES, false);
+        test_modify_group(&rsclient, &DEFAULT_NOT_HP_GROUP_NAMES, true);
+
+        rsclient.idm_group_create("test_group").unwrap();
+        rsclient
+            .idm_group_add_members("test_group", vec!["test"])
+            .unwrap();
+        assert!(is_attr_writable(&rsclient, "test_group", "description").unwrap());
+    });
+}
+
+// Admins
+// read and write access control entries.
+#[test]
+fn test_default_entries_rbac_admins_access_control_entries() {
+    run_test(|rsclient: KanidmClient| {
+        rsclient
+            .auth_simple_password("admin", ADMIN_TEST_PASSWORD)
+            .unwrap();
+        static ACP_COMMON_ATTRS: [&str; 4] =
+            ["name", "description", "acp_receiver", "acp_targetscope"];
+        static ACP_ENTRIES: [&str; 28] = [
+            "idm_admins_acp_recycle_search",
+            "idm_admins_acp_revive",
+            "idm_self_acp_read",
+            "idm_self_acp_write",
+            "idm_all_acp_read",
+            "idm_acp_people_read_priv",
+            "idm_acp_people_write_priv",
+            "idm_acp_people_manage",
+            "idm_acp_people_account_password_import_priv",
+            "idm_acp_people_extend_priv",
+            "idm_acp_group_write_priv",
+            "idm_acp_account_read_priv",
+            "idm_acp_account_write_priv",
+            "idm_acp_account_manage",
+            "idm_acp_radius_servers",
+            "idm_acp_hp_account_read_priv",
+            "idm_acp_hp_account_write_priv",
+            "idm_acp_hp_group_write_priv",
+            "idm_acp_schema_write_attrs_priv",
+            "idm_acp_acp_manage_priv",
+            "idm_acp_schema_write_classes_priv",
+            "idm_acp_group_manage",
+            "idm_acp_hp_account_manage",
+            "idm_acp_hp_group_manage",
+            "idm_acp_domain_admin_priv",
+            "idm_acp_system_config_priv",
+            "idm_acp_account_unix_extend_priv",
+            "idm_acp_group_unix_extend_priv",
+        ];
+
+        ACP_ENTRIES.iter().for_each(|entry| {
+            test_read_attrs(&rsclient, entry, &ACP_COMMON_ATTRS, true);
+            test_write_attrs(&rsclient, entry, &ACP_COMMON_ATTRS, true);
+        });
+    });
+}
+
+// read schema entries.
+// TODO: write schema entries
+#[test]
+fn test_default_entries_rbac_admins_schema_entries() {
+    run_test(|rsclient: KanidmClient| {
+        rsclient
+            .auth_simple_password("admin", ADMIN_TEST_PASSWORD)
+            .unwrap();
+        let default_classnames: HashSet<String> = [
```

Comment:
You could beild this set from the actual schema rather than hardcoding? But maybe that needs schema apis to be added ....

Reply:
I don't know how to get all pub static from constants module.

Reply:
Ahhh that's not what I mean. Schema is loaded into the database as entries, so there is a rest api here for it:

https://github.com/kanidm/kanidm/blob/master/kanidmd/src/lib/core/mod.rs#L1500

So you could have the client call that api, then you can create the "all possible classnames" from the live running schema of the server. Does that make sense?

Reply:
I test against that response. See line L418. Am I missing something?

Reply:
OHHHH I see. Okay, I missed this. All good then!